crates/task-impls/src/helpers.rs

```diff
@@ -499,11 +484,14 @@ pub fn validate_proposal_view_and_certs<TYPES: NodeType>(
         proposal.data.clone()
     );
 
-    let view_leader_key = quorum_membership.leader(view);
-    ensure!(
-        view_leader_key == *sender,
-        "Leader key does not match key in proposal"
-    );
+    // Validate the proposal's signature. This should also catch if the leaf_commitment does not equal our calculated parent commitment
+    //
+    // There is a mistake here originating in the genesis leaf/qc commit. This should be replaced by:
```

Comment:
Looks like we might be able to remove this comment entirely? Looks like we can keep the above, but the "there is a mistake here" part seems to be rectified.